Sources/NIOPosix/VsockAddress.swift

```diff
@@ -0,0 +1,256 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+#if canImport(Darwin) || os(Linux)
```

Comment:
@weissi could you advise how we should structure these changes to safely not break builds on other platforms?

This file contains the definition of the `VsockAddress` type, which is used in the new public APIs, e.g. `Bootstrap.connect(to vsockAddress:)`. If we remove the guard from the public API, then `VsockAddress` also needs to be unconditionally available.

However, this file also contains the various extensions required on other types, e.g. `NIOBSDSocket`, `ProtocolFamily`, `AddressFamily`, `sockaddr_vm`, `sockaddr_storage`, `BaseSocket`, and `ChannelOptions`.

If I just remove the `#if canImport(Darwin) || or os(Linux) from this line, then I expect it will just fail to compile on other platforms.

Some of these require the C library be present, which is why this file has the (almost) usual import dance. What it does _not_ currently have is the `#error` I see in other places:

```swift
#error("The ... module was unable to identify your C library.")
```

Should we be having this `#error` here and do I need to expand the import dance to include Windows, even though we will probably throw an error in the  body of the APIs?

I'm a bit nervous about breaking the build for other platforms.

Reply:
Everything but `sockaddr_vm` should be available on all supported platforms. For `extension sockaddr_vm` I think it's fine to put an `#if` guard around that.